source/agora/consensus/EnrollmentPool.d

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************************
+
+    Contains supporting code for managing enrollments registered by nodes
+    on the network to be a validator, which means the enrollment information
+    needs to be confirmed on the consensus protocol to be a validator. The
+    information is stored in a table of SQLite.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.EnrollmentPool;
+
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.consensus.data.Block;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.PreimageInfo;
+import agora.consensus.data.UTXOSet;
+import agora.consensus.Validation;
+import agora.utils.Log;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+mixin AddLogger!();
+
+/// Ditto
+public class EnrollmentPool
+{
+    /// SQLite DB instance
+    private Database db;
+
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            db_path = path to the database file, or in-memory storage if
+                        :memory: was passed
+
+    ***************************************************************************/
+
+    public this (string db_path)
+    {
+        this.db = Database(db_path);
+
+        // create the table for enrollment pool if it doesn't exist yet
+        this.db.execute("CREATE TABLE IF NOT EXISTS enrollment_pool " ~
```

Comment:
Do we really need a table here ?

Reply:
If we don't need this new table, I didn't need to start to resolve this issue. This PR and the issue related to this is all about creating a new table like enrollment_pool.

Reply:
> No checking the block, no requesting other nodes.

You're right. We need some recovery code like requesting enrollments from other nodes. We need another issue for that and I'm sorry for the late catch.

Reply:
I have created an issue for the problem. #755